hoodie-client/src/main/java/com/uber/hoodie/config/HoodieWriteConfig.java

```diff
@@ -102,6 +105,17 @@ public StorageLevel getWriteStatusStorageLevel() {
         return StorageLevel.fromString(props.getProperty(WRITE_STATUS_STORAGE_LEVEL));
     }
 
+    public <T extends WriteStatus> T getWriteStatusInstance() {
+        try {
+            Class<? extends WriteStatus> aClass = (Class<? extends WriteStatus>) Class
```

Comment:
Use ReflectionUtils to cache the class instance and config is not the right place to instantiate the class. I would do it as a interface method (factory) in WriteStatus interface

Reply:
Sure Prasanna, will do.

Reply:
I agree with what prasanna is saying. We can't be actually instantiating the class from the config class.  i.e the method here can just return you the class name.. 

Please move the actual creation of the class i.e the ReflectionUtils call into the Handle classes themselves.